scripts/index_voices.py

```
@@ -0,0 +1,1418 @@
+
+            ))
+
+    def get_piper_voice_list(self):
+        """
+        Fetches the Piper voices manifest from the Rhasspy piper-voices repository and registers each voice in the manager.
+
+        Downloads the voices.json manifest, creates a TTSModelInfo for each entry (deriving a voice_id prefixed with "piper_", a standardized language tag, and the first ONNX and JSON file URLs from the entry), and calls add_voice to store it. If an entry cannot be processed, prints a failure message for that voice.
+        """
+        base = "https://huggingface.co/rhasspy/piper-voices/resolve/main/"
+        voice_list = "https://huggingface.co/rhasspy/piper-voices/resolve/main/voices.json"
+        piper_voices = requests.get(voice_list).json()
```

Ruff (0.14.5)

Probable use of `requests` call without timeout

(S113)